### ISSUE TITLE
How do I get the parameter.json or ism7config tool to create it.

### ISSUE THREAD

**Issue by NihilUnum** [zivillian/ism7mqtt#97]
Hi, 

I wanted to use the Docker version of ism7mqtt with Docker Compose. Here is my docker-compose.yml file:

```

version: "3.7"
services:
  ism7mqtt:
    image: zivillian/ism7mqtt:latest
    container_name: ism7mqtt
    restart: unless-stopped
    volumes:
      - ./parameter.json:/app/parameter.json
    environment:
      - ISM7_IP=192.168.11.250
      - ISM7_PASSWORD="PASSWORD"
      - ISM7_MQTTHOST=192.168.11.2
      - ISM7_MQTTUSERNAME=MQTT_USER
      - ISM7_MQTTPASSWORD=PASSWORD
```

However, I'm having issues creating the parameter.json file. You mentioned that one must use the ism7config tool, which is now inside the Docker container. My problem is that I can't access it because Docker won't start without the parameter.json file. If I create an empty parameter.json file, it fails with an error 

```
Attaching to ism7mqtt
ism7mqtt  | System.Text.Json.JsonException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. Path: $ | LineNumber: 0 | BytePositionInLine: 0.
ism7mqtt  |  ---> System.Text.Json.JsonReaderException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. LineNumber: 0 | BytePositionInLine: 0.
ism7mqtt  |    at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan`1 bytes)
ism7mqtt  |    at System.Text.Json.Utf8JsonReader.Read()
ism7mqtt  |    at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
ism7mqtt  |    --- End of inner exception stack trace ---
ism7mqtt  |    at System.Text.Json.ThrowHelper.ReThrowWithPath(ReadStack& state, JsonReaderException ex)
ism7mqtt  |    at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
ism7mqtt  |    at System.Text.Json.JsonSerializer.ReadFromSpan[TValue](ReadOnlySpan`1 utf8Json, JsonTypeInfo jsonTypeInfo, Nullable`1 actualByteCount)
ism7mqtt  |    at System.Text.Json.JsonSerializer.ReadFromSpan[TValue](ReadOnlySpan`1 json, JsonTypeInfo jsonTypeInfo)
ism7mqtt  |    at System.Text.Json.JsonSerializer.Deserialize[TValue](String json, JsonSerializerOptions options)
ism7mqtt  |    at ism7mqtt.Ism7Config..ctor(String filename) in /app/ism7mqtt/ISM7/Ism7Config.cs:line 32
ism7mqtt  |    at ism7mqtt.Ism7Client..ctor(Func`3 messageHandler, String parameterPath, IPAddress ipAddress) in /app/ism7mqtt/ISM7/Ism7Client.cs:line 44
ism7mqtt  |    at ism7mqtt.Program.Main(String[] args) in /app/ism7mqtt/Program.cs:line 125
ism7mqtt  | Unhandled exception. System.Text.Json.JsonException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. Path: $ | LineNumber: 0 | BytePositionInLine: 0.
ism7mqtt  |  ---> System.Text.Json.JsonReaderException: The input does not contain any JSON tokens. Expected the input to start with a valid JSON token, when isFinalBlock is true. LineNumber: 0 | BytePositionInLine: 0.
ism7mqtt  |    at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan`1 bytes)
ism7mqtt  |    at System.Text.Json.Utf8JsonReader.Read()
ism7mqtt  |    at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
ism7mqtt  |    --- End of inner exception stack trace ---
ism7mqtt  |    at System.Text.Json.ThrowHelper.ReThrowWithPath(ReadStack& state, JsonReaderException ex)
ism7mqtt  |    at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
ism7mqtt  |    at System.Text.Json.JsonSerializer.ReadFromSpan[TValue](ReadOnlySpan`1 utf8Json, JsonTypeInfo jsonTypeInfo, Nullable`1 actualByteCount)
ism7mqtt  |    at System.Text.Json.JsonSerializer.ReadFromSpan[TValue](ReadOnlySpan`1 json, JsonTypeInfo jsonTypeInfo)
ism7mqtt  |    at System.Text.Json.JsonSerializer.Deserialize[TValue](String json, JsonSerializerOptions options)
ism7mqtt  |    at ism7mqtt.Ism7Config..ctor(String filename) in /app/ism7mqtt/ISM7/Ism7Config.cs:line 32
ism7mqtt  |    at ism7mqtt.Ism7Client..ctor(Func`3 messageHandler, String parameterPath, IPAddress ipAddress) in /app/ism7mqtt/ISM7/Ism7Client.cs:line 44
ism7mqtt  |    at ism7mqtt.Program.Main(String[] args) in /app/ism7mqtt/Program.cs:line 125
ism7mqtt  |    at ism7mqtt.Program.<Main>(String[] args)
ism7mqtt exited with code 0
```

If I try to start it without the parameter.json file, it creates a directory named parameter.json instead and fails to start with the message:

```
Attaching to ism7mqtt
ism7mqtt  | 'parameter.json' does not exist
ism7mqtt exited with code 0
```

I spent some time figuring out that ism7mqtt doesn't like "!" in the MQTT broker's password. I was able to change that relatively quickly. This might be helpful for someone else; using Password! or 'Password!' seems to block the authentication process to the MQTT broker somewhere. Unfortunately, I don't have the exact error message anymore. 

I can connect without any issues using a mobile phone or a Windows app, and I can access smartsetpc.sdf, but how do I get a .json file out of it? 
I'm currently stuck and could use a nudge in the right direction.

Thank you for your app and in advance for your help

best
Adrian

**Comment by NihilUnum** [comment]
Alright, I managed to launch my Docker container using a dummy parameter.json file and with the command 
`sudo docker compose exec ism7mqtt sh`
 Additionally, I was able to initiate ism7config using 
`ism7config -p 192.168.11.250 -i password`
but encountered a timeout. My assumption is that ism7mqtt was connected to ism, and from what I understand, only one connection is possible at a time. 
Fortunately, I discovered this [Releases](https://github.com/zivillian/ism7mqtt/releases) link. It seems I was just too blind to notice it yesterday.
Now I can try to get my own parameter.json with ism7parameter for Windows/Linux.
